File: php/hamming/hamming.php
<?php

//
// This is only a SKELETON file for the "Hamming" exercise. It's been provided as a
// convenience to get you started writing code faster.
// Completed by Sofia Nieves
// Mon 30 May 04:56:34 UTC 2016
//

function distance($a, $b)
{
    $distance = 0;

    if (strlen($a) != strlen($b))
    {
        throw new InvalidArgumentException('DNA strands must be of equal length.');
    }

    for ($i = 0; $i < strlen($a); $i++)
    {
        if (substr($a, $i, 1) != substr($b, $i, 1))
        {
            $distance++;
        }
    }

    return $distance;
}

File: python/hamming/hamming.py
#
# Solution for hamming
# Written by Sofia Nieves
# Thu 16 Jun 05:27:50 UTC 2016
#

def distance(strand1, strand2):
    # Return nonsensical number if strands are different lengths
    if len(strand1) != len(strand2):
        return -1

    distance = 0

    for position, nucleotide in enumerate(strand1):
        if strand1[position] != strand2[position]:
            distance += 1

    return distance
